documentation/docs/11-rendering-and-routing.md

```diff
@@ -0,0 +1,77 @@
+---
+title: Rendering and routing
+---
+
+By default, SvelteKit will server-render pages on-demand, then **hydrate** the rendered HTML in the client with an interactive Svelte app while initialising a **router** that takes over subsequent navigations.
+
+You can control each of these on a per-app or per-page basis. Note that each of the per-page settings use [`context="module"`](https://svelte.dev/docs#script_context_module), and only apply to page components, _not_ [layout](#layouts) components.
+
+If both are specified, per-page settings override per-app settings in case of conflicts. Each setting can be controlled independently, but `ssr` and `hydrate` cannot both be `false` since that would result in nothing being rendered at all.
+
+### ssr
+
+Disabling server-side rendering effectively turns your SvelteKit app into a **single-page app** or SPA.
+
+> In most situations this is not recommended: it harms SEO, tends to slow down perceived performance, and makes your app inaccessible to users if JavaScript fails or is disabled (which happens [more often than you probably think](https://kryogenix.org/code/browser/everyonehasjs.html)). Sometimes it's appropriate or even necessary, but consider alternatives before disabling SSR.
+
+You can disable SSR app-wide with the [`ssr` config option](#configuration-ssr), or a page-level `ssr` export:
+
+```html
+<script context="module">
+	export const ssr = false;
```

Comment:
could this be a `function`? e.g. I previously wanted to do [dynamic rendering](https://developers.google.com/search/docs/guides/dynamic-rendering) where I give an `ssr` version to google and a SPA version for everyone else. the function would need a context (or milieu if you prefer) taking the current request (or at least request headers)

you can see what I'd proposed in Sapper here: https://github.com/sveltejs/sapper/pull/1275/files

Reply:
Oh, interesting. As far as the technical implementation goes that would be very straightforward I think, but you probably want an option like that to apply at the app level rather than the page level. Any thoughts on what a good app-level API for that would look like?

Reply:
To detect the user agent we'd need the request. Not sure if that's available at the app level?

Either way, I was thinking the API might look something like `ssr: function({ ...request })` or `ssr: function({ headers })`

Reply:
It's available in `handle`, but that function signature would have to change and I'm not sure how

Reply:
if we decide to do this it can be in a follow-up that changes the signature of the `render` function passed to `handle`